src/pages/FavoriteWatch/FavoriteWatch.page.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useCallback, useMemo } from 'react';
+
+import { useLocation } from 'react-router-dom';
+import { useAuthContext } from '../../providers/Auth';
+
+import Styled from '../Watch/Watch.styled';
+import Loading from '../../components/Common/Loading';
+import GridVideo from '../../components/GridVideo';
+import VideoDetail from '../../components/VideoDetail/VideoDetail.component';
+
+import { withPageLayout } from '../../components/Layout';
+import { useVideoContext } from '../../providers/Video';
+
+const FavoriteWatch = () => {
+  const { search } = useLocation();
+  const {
+    state: { video, isLoading },
+    actions: { fetchVideoDetail },
+  } = useVideoContext();
+
+  const {
+    state: { favorites, isAuthenticated },
+    actions: { handleAddToFavorites, handleRemoveFromFavorites },
+  } = useAuthContext();
+
+  const videoId = useMemo(() => {
+    const id = search.replace('?v=', '');
+
+    return id;
+  }, [search]);
+
+  const handleFetchVideoDetail = useCallback(async () => {
```

Comment:
I think `useCallback` does nothing here, from Reactjs docs:

`useCallback` will return a memoized version of the callback that only changes if one of the dependencies has changed. This is useful when **passing callbacks to optimized child components** that rely on reference equality to prevent unnecessary renders

Reply:
Ok I got it. Thank you for your feedback!